plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts

```diff
@@ -119,6 +121,181 @@ export function createRouter({
     },
   );
 
+  router.post('/createArgo', async (request, response) => {
+    const argoInstanceName = request.body.clusterName;
+    const namespace = request.body.namespace;
+    const projectName = request.body.projectName as string;
+    const appName = request.body.appName as string;
+    const labelValue = appName;
+    const sourceRepo = request.body.sourceRepo;
+    const sourcePath = request.body.sourcePath;
+    const argoSvc = new ArgoService(argoUserName, argoPassword, config);
```

Comment:
Now that we are starting to have more of these endpoints in this router, would it make sense to initialize this `ArgoService` outside of the endpoint itself and reuse the same instance on all of them?

Reply:
100%. changing